src/app/components/fhirdata/fhirdata.component.ts

```diff
@@ -145,37 +149,39 @@ export class FhirdataComponent implements OnInit {
   }
 
   fetchPatientDataInSession(typeOfQueryFlag?: string) {
-    let query = this.utilService.queryString('Patient', `?_id=${this.returnPatientId()}&_revinclude:iterate=ExplanationOfBenefit:patient`);
+    let query = this.utilService.queryString('Patient', `?_id=${this.utilService.returnPatientId()}&_revinclude:iterate=ExplanationOfBenefit:patient`);
 
     if (typeOfQueryFlag === "everything") {
-      query = this.utilService.queryString('Patient', `/${this.returnPatientId()}/$everything`);
+      query = this.utilService.queryString('Patient', `/${this.utilService.returnPatientId()}/$everything`);
     }
 
     this.searchTypeHeader = `${environment.fhirEndpointUri}${query}`;
-    this.apiCallFunction(query, this.headers());
+    this.apiCallFunction(query, this.httpService.getHeaders());
   }
 
   private apiCallFunction(query: string, headers?: HttpHeaders) {
-    if (this.oauthService.hasValidAccessToken()) {
-      this.showLoadingBar = true;
-      this.utilService.resetErrorObject(this.errorObject);
-      this.httpService.getFhirQueries(query, headers).subscribe(
-        recordsFound => {
-          const result = this.setupDataInView(recordsFound);
-          if (result === 'dataFound') {
-            this.bundleFound = recordsFound;
+    this.oidcSecurityService.isAuthenticated().subscribe(isAuthenticated => {
+      if (isAuthenticated) {
+        this.showLoadingBar = true;
```

Comment:
move this statements inside if conditions to a method of its own for better readability.

Reply:
Migrated to a separate private function in the component